troposphere/emr.py

```diff
@@ -259,18 +259,46 @@ class InstanceGroupConfigProperty(AWSProperty):
         'Name': (basestring, False),
     }
 
+class OnDemandProvisioningSpecification(AWSProperty):
+    props = {
+        'AllocationStrategy': (basestring, True),
+    }
+
+    def validate(self):
+        valid_values = [ 'lowest-price' ]
+
+        allocation_strategy = self.properties.get('AllocationStrategy', None)
+
+        if allocation_strategy not in valid_values:
+            raise ValueError(
+                'Only lowest-price is a valid AllocationStrategy'
```

Comment:
should have more general messages so they don't have to be changed when `valid_values` expands

Reply:
Good remark. I changed the error message. It now also prints the provided invalid AllocationStrategy.